node/src/service.rs

```diff
@@ -239,11 +239,182 @@ async fn build_relay_chain_interface(
 
 // We use this as the RuntimeApi definition in the following function so that we don't need to specify so many generic traits.
 // This might have to change if we want to use this function for other runtimes.
-use foucoco_runtime::RuntimeApi as FoucocoRuntimeApi;
+use amplitude_runtime::RuntimeApi as AmplitudeRuntimeApi;
+
+/// Start a node that exposes the RPC endpoints for the spacewalk pallets for the amplitude runtime.
+#[sc_tracing::logging::prefix_logs_with("Parachain")]
+async fn start_node_impl_spacewalk_amplitude<Executor>(
```

Comment:
This quite complex function has been duplicated now three times.

The first time it has been duplicated from [`start_node_impl`](https://github.com/pendulum-chain/pendulum/blob/2783abf0a2abe70c3e7258c4aef2941681789fc0/node/src/service.rs#L587) in PR https://github.com/pendulum-chain/pendulum/pull/153 although the only difference was a single line (call `crate::rpc::create_full` vs `crate::rpc::create_full_spacewalk`).

This duplicate did not use the generic parameter `RuntimeApi` so now it needed to be duplicated again.

But this will be painful to maintain correctly with the next Polkadot version updates...

Can we not just use the one function and just add a single Boolean parameter to distinguish between `create_full` and `create_full_spacewalk`?

Reply:
I updated the [initial description of the issue ticket](https://github.com/pendulum-chain/pendulum/pull/116#issue-1582273849). Did you see that before writing this comment? 

I agree that this is ugly and we should get rid of it. But it seems quite heavy to figure the generic types out and I fear we don't have the time to do that at the moment. If you feel confident that you can figure it out quickly that's great.

Reply:
I actually did not read it. So I agree, let's keep it like this right now.

The trait bounds are painful, I agree, and it is one of the pain points of previous Polkadot version updates. Even if I would be able to fix this now, let's not bother with it.

Resolved.

Reply:
link to issue: https://github.com/pendulum-chain/pendulum/issues/161